debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ChangeStreamPipelineFactory.java

```diff
@@ -188,36 +239,48 @@ private static Optional<Bson> createOperationTypeFilter(MongoDbConnectorConfig c
             includedOperations.remove(OperationType.DELETE);
         }
 
-        return Optional.of(Filters.in("event.operationType", includedOperations.stream()
+        var field = filterConfig.isLiteralsMatchMode()
+                ? "operationType"
+                : "event.operationType";
```

Comment:
Why is this different? Is it because one is downstream in the pipeline? A comment would be useful.

Reply:
Added a comment -- although from the entire code it was quite clear that when regexes are used the original change stream documment is embedded into a new document `{namespace: <fqid>, event: <orginal>}`